controllers/contact-controller.js

```diff
@@ -308,11 +321,24 @@ const ContactController = Contact => {
     };
 
     const contact = await Contact.create(contactRecord);
+
+    // Link all of the csam reports related to this contact
+    if (body.csamReports && body.csamReports.length) {
+      await Promise.all(
+        body.csamReports.map(r =>
+          CSAMReportController.connectToContact(contact.id, r.id, accountSid),
+        ),
+      );
+    }
```

Comment:
Note that this can fail if the `csamReports` contains objects with ids that does not correlates with a csam report record for this particular account sid, hence the entire "create contact" will finish with failure. What do you think will be best way to handle this (unlikely but possible) scenario? Try to do so and continue in case of failure, returning success to the user based on the contact creation status is what makes sense to me. What do you think?